Better Management of WIT and caller-utils generation logs.

Gohlub opened the pull request:
## Problem

Output after running 'kit b --hyperapp' is too verbose (everything was handled by println! macros). Slightly hard to understand and clogs the LLM context unnecessarily.

## Solution

Updated print statements into their respective categories. If dev/llm now wants to access detailed logs of WIT and caller utils generation, should just use: `RUST_LOG=debug kit b --hyperapp`

## Docs Update
TODO
[Corresponding docs PR](https://github.com/hyperware-ai/hyperdrive-book/pull/my-pr-number)

## Notes
when running `RUST_LOG=debug kit b --hyperapp`, things show up as expected, but this particular log seems to display a lot of times, which makes other log entries harder to see. I suggest trying to run it with some of your own projects and see whether the output includes what we want it to.
```bash
...
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 532409368 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 544341034 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 523536015 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 540541841 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 535951726 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 747166189 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 745921931 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 749082619 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 764302100 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 764302100 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 545426833 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 537555821 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 545426833 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 595754349 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 606366714 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 606366714 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 539732669 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 536238603 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 551004872 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 527804670 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 644223684 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 647818668 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 647818668 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 647818668 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 647818668 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 545385125 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 545385125 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 554878316 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 519321069 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 565135303 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 569810669 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 952996282 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 730466157 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 550135366 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 530950275 }), most_recent_excluded: None
DEBUG src/build/mod.rs:674: get_most_recent_modified_time: most_recent: Some(SystemTime { tv_sec: 1745618203, tv_nsec: 523293847 }), most_recent_excluded: None
...
```
this here is just a snippet, there's a lot more. My goal was to have these logs available to the LLM should there be any issues, but this would only add garbage to the context.

nick1udwig commented on src/build/wit_generator.rs:
```diff
@@ -846,24 +857,23 @@ fn process_rust_project(project_path: &Path, api_dir: &Path) -> Result<Option<(S
                     }
                 } else {
-                    println!("    Skipping method without relevant attributes");
+                    warn!("    Method {} does not have the [remote], [local], or [http], and will not be included in the WIT file", method_name);
```
This prints
```
    Method init does not have the [remote], [local], or [http], and will not be included in the WIT file
```
which is on by default for output to stdout. Should we demote to `info`? Should we special-case an `#[init]` flagged method? It seems inappropriate to output every build

Gohlub replied:
Solved this by handling the special #[init] case, it should now warn only if there is any other function attribute other than the 4 allowed ones.

nick1udwig commented on src/build/wit_generator.rs:
```diff
@@ -673,14 +685,15 @@ impl AsTypePath for syn::Type {
+#[instrument(level = "trace", skip_all)]
 fn process_rust_project(project_path: &Path, api_dir: &Path) -> Result<Option<(String, String)>> {
-    println!("\nProcessing project: {}", project_path.display());
+    info!("Processing project: {}", project_path.display());
```
This should probably be `debug`

nick1udwig commented on src/build/caller_utils_generator.rs:
```diff
@@ -82,6 +84,7 @@ fn find_world_names(api_dir: &Path) -> Result<Vec<String>> {
 }
 
 // Convert WIT type to Rust type - IMPROVED with more Rust primitives
+#[instrument(level = "trace", skip_all)]
```
We only need these decorators for functions that return a `Result`: get rid of them on all the non-`Result` returning functions

(What they do is they give us an output trace in case an error or crash occurs)

nick1udwig reviewed (approved):
Looks great! Please address comments and then merge away

Hadn't seen the multi-arg tracing syntax before: TIL. I noticed you applied it in caller_utils_generator.rs but not in wit_generator.rs. Any reason here? I don't have a strong feeling that it must be applied in both, given we don't apply it in general, just curious why.

Regarding `most_recent_excluded` spam, please do get rid of that `debug!()` line.

Gohlub commented:
Just to consolidate what I changed:
- Consolidated the tracing macros to follow the same conventions
- Added special case handling for when #[init] an attribute in the impl block of a process
- got rid of the instrument decorator for functions that don't return a Result
- changed the 'entry' log on process_rust_project (see comment above) to be debug instead of info

nick1udwig reviewed (approved):
Nice work :zap:
